nservicebus/handlers/index.md

```diff
@@ -43,6 +43,22 @@ Receiving a message for which there are no message handlers is considered an err
 
 partial: behaviorcaveat
 
+## Multiple handlers
+
+When an endpoint hosts multiple matching handlers for a single incoming message and one of the handlers fails then the incoming message gets retried. When the incoming message is retried all matching handlers get invoked again. Invoked handlers include the handlers that might had already been succesfully invoked in previous attempts.
+
+In general, an incoming message represents a single unit of work. If multiple handlers get invoked it is best that either all succeed or all fail in a single transactional boundary. It is not recommended to run multiple atomic or isolated transactional operations for a single incoming message.
+
+Ensure correct invocation of isolated or shared state by applying one of more of the strategies:
+
+- Ensuring all handlers share the same transactional context to ensure any the shared transaction can rollback any operations (connection sharing, MSDTC, etc.).
+- Ensuring handlers are idempotent and correctly deal with At-least-once Delivery.
+- Hosting each handler in its own endpoint (hosting in isolation) so each subscribers receive its own own copy to process in isolation (multiple subscibers).
+- When receiving the message do a `SendLocal` with a new message but same payload so that each task gets represented by its own message and can be executed in isolation. (single subscriber, split at the target).
```

Comment:
```suggestion
- Split the message at the target - do `SendLocal` with a new message but the same payload so that each task gets represented by its own message and gets executed in isolation. 
```